File: data/code_pairs/cpp/2175.cpp
#include <iostream>
#include <cstring>
#include <cstdlib>

struct Block {
    int d, e;
    float f;
} block = {42, 43, 2.0f};

struct Block2 {
    char s[16];
} block2 = {"Hello World    "};

void test() {
    if (block.d != 42 || block.e != 43 || block.f != 2.0f) {
        std::cerr << "Test failed due to incorrect values in block" << std::endl;
        std::exit(1);
    }
    if (std::strncmp(block2.s, "Hello World    ", 15) != 0) {
        std::cerr << "Test failed due to incorrect value in block2" << std::endl;
        std::exit(1);
    }
    std::cout << "All tests passed." << std::endl;
}

int main() {
    test();
    return 0;
}

File: data/code_pairs/fortran/2175.f90
module common_blocks
  implicit none
  character(len=15) :: c
  integer :: d, e
  real :: f
end module common_blocks

subroutine test()
  use common_blocks
  if ((d .ne. 42) .or. (e .ne. 43) .or. (f .ne. 2.0)) then
    print *, "Test failed due to incorrect values in /block/"
    stop
  endif
  if (c .ne. "Hello World    ") then
    print *, "Test failed due to incorrect value in /block2/"
    stop
  endif
  print *, "All tests passed."
end subroutine test

program main
  use common_blocks
  d = 42
  e = 43
  f = 2.0
  c = "Hello World    "
  call test()
end program main